### PR TITLE
DD4hep: Legacy DD Shape ID Support and Mapping

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Add mapping between the old DD and the new DD4hep shape identifiers
* Add unit test to check the conversion
* Add a convenience ``` LegacySolidShape legacyShape(const cms::DDSolidShape shape) const``` function to request an old shape ID from the ```cms::DDFilteredView```

Example:
```
LegacySolidShape legacySphere = cms::dd::value(cms::LegacySolidShapeMap, cms::DDSolidShape::ddsphere);
```

#### PR validation:

Run unit test:
```
testDD4hepDDSolidLegacyShapes
```

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by cvuosalo** [comment]
Please don't merge this PR yet. It's only four hours old, and I haven't had a chance to look at it.

File: DetectorDescription/DDCMS/data/cms-mf-geometry.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
     <Include ref='Geometry/CMSCommonData/data/cms.xml'/>
     <Include ref='DetectorDescription/DDCMS/data/cmsMagneticField.xml'/>
     <Include ref='MagneticField/GeomBuilder/data/MagneticFieldVolumes_160812_1.xml'/>
+    <Include ref='MagneticField/GeomBuilder/data/MagneticFieldVolumes_160812_2.xml'/>
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/data/cms-mf-geometry.xml]: Why is this file in this PR? It has nothing to do with shapes.

Comment (ianna) [reply DetectorDescription/DDCMS/data/cms-mf-geometry.xml]: @cvuosalo - this is to complete the MF test.

File: DetectorDescription/DDCMS/interface/DDSolidShapes.h [hunk 2]
```diff
@@ -77,6 +99,31 @@ namespace cms {
...
+      {{LegacySolidShape::dd_not_init, cms::DDSolidShape::dd_not_init},
+       {LegacySolidShape::ddbox, cms::DDSolidShape::ddbox},
+       {LegacySolidShape::ddtubs, cms::DDSolidShape::ddtubs},
+       {LegacySolidShape::ddtrap, cms::DDSolidShape::ddtrap},
+       {LegacySolidShape::ddcons, cms::DDSolidShape::ddcons},
+       {LegacySolidShape::ddpolycone_rz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolycone_rz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolycone_rrz, cms::DDSolidShape::ddpolycone},
+       {LegacySolidShape::ddpolyhedra_rz, cms::DDSolidShape::ddpolyhedra},
+       {LegacySolidShape::ddpolyhedra_rz, cms::DDSolidShape::ddpolyhedra},
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/interface/DDSolidShapes.h]: Why are there duplicate lines? If they are not a mistake, they need a comment explaining them so I don't delete them some time in the future. What is the significance of the number 22?

Comment (ianna) [reply DetectorDescription/DDCMS/interface/DDSolidShapes.h]: @cvuosalo - ```std::array``` is a container that encapsulates fixed size arrays. If you change this number by mistake the compiler will complain :-)

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
@cvuosalo so what is you conclusion, based on the latest updates by @ianna ?

**Comment by cvuosalo** [comment]
@fabiocos: I see the problem was fixed. I say go forward with this PR.

**Comment by fabiocos** [comment]
+1

**Pull request merged**